### ISSUE TITLE
dd-trace throws error, cannot find module 'retry'

### ISSUE THREAD

**Issue by gaggle** [DataDog/dd-trace-js#1623]
**Describe the bug**
I see this error on starting my app, coming from dd-trace:
```
Error: Cannot find module 'retry'
Require stack:
- /app/node_modules/.pnpm/dd-trace@1.4.0/node_modules/dd-trace/packages/dd-trace/src/profiling/exporters/agent.js
- /app/node_modules/.pnpm/dd-trace@1.4.0/node_modules/dd-trace/packages/dd-trace/src/profiling/config.js
- /app/node_modules/.pnpm/dd-trace@1.4.0/node_modules/dd-trace/packages/dd-trace/src/profiling/profiler.js
- /app/node_modules/.pnpm/dd-trace@1.4.0/node_modules/dd-trace/packages/dd-trace/src/profiling/index.js
- /app/node_modules/.pnpm/dd-trace@1.4.0/node_modules/dd-trace/packages/dd-trace/src/profiler.js
- /app/node_modules/.pnpm/dd-trace@1.4.0/node_modules/dd-trace/packages/dd-trace/src/proxy.js
- /app/node_modules/.pnpm/dd-trace@1.4.0/node_modules/dd-trace/packages/dd-trace/index.js
- /app/node_modules/.pnpm/dd-trace@1.4.0/node_modules/dd-trace/index.js
- /app/apps/pdf-gen/build/logging/tracing.js
- /app/apps/pdf-gen/build/logging/index.js
- /app/apps/pdf-gen/build/app.js
- /app/apps/pdf-gen/build/server.js
```

I notice `https://github.com/DataDog/dd-trace-js/blob/7ffe1fb7dfe01d2ce0d1108406de25ef84557004/packages/dd-trace/src/profiling/exporters/agent.js` is importing "retry":
```
const retry = require('retry')
```

Is there awareness that `retry` is a devDependency? If `exporters/agent.js` is expected to be used at runtime then it looks to me retry should be made a full dependency? Or am I missing something?

**Environment**
Node 16

* **Operation system:** MacOS
* **Node version:** Node 16
* **Tracer version:**: ^1.3.0
* **Agent version:**: 7

**Comment by rochdev** [comment]
For now I opened #1624 to fix the immediate issue. We'll be adding tests to isolate a version of dd-trace without dependencies in another PR.

**Comment by rochdev** [comment]
Fixed in 1.4.1
